frontend/src/lib/components/flows/flowStore.ts

```diff
@@ -0,0 +1,151 @@
+
+export const isCopyFirstStepSchemaDisabled = derived(flowStore, (flow: Flow) => {
+	const modules = flow.value.modules
+	const [firstModule] = modules
+	return (
+		modules.length === 0 || (firstModule.value.path === '' && firstModule.value.type === 'script')
+	)
+})
+
+export function addModule() {
+	const newModule: FlowModule = {
+		value: { type: FlowModuleValue.type.SCRIPT, path: '' },
+		input_transform: {}
+	}
+
+	flowStore.update((flow: Flow) => {
+		flow.value.modules = flow.value.modules.concat(newModule)
+		return flow
+	})
+}
+
+export async function pickScript(path: string, step: number) {
+	flowStore.update((flow: Flow) => {
+		if (flow.value.modules[step]) {
+			flow.value.modules[step].value.path = path
+		}
+
+		return flow
+	})
+
+	await loadSchema(step)
+}
+
+export async function createInlineScriptModule(language: FlowModuleValue.language, step: number) {
+	const code = language === FlowModuleValue.language.DENO ? DENO_INIT_CODE : PYTHON_INIT_CODE
+	flowStore.update((flow: Flow) => {
+		flow.value.modules[step].value = {
+			type: FlowModuleValue.type.RAWSCRIPT,
+			content: code,
+			language
+		}
+
+		return flow
+	})
+	await loadSchema(step)
+}
+
+export async function loadSchema(step: number) {
+	const flow = get(flowStore)
+	const module = flow.value.modules[step]
+
+	const { input_transform, schema } = await loadSchemaFromModule(module)
+
+	flowStore.update((flow: Flow) => {
+		flow.value.modules[step].input_transform = input_transform
+		return flow
+	})
+	schemasStore.update((schemas) => {
+		schemas[step] = schema
+		return schemas
+	})
+}
+
+export async function fork(step: number) {
+	const flow = get(flowStore)
+	const flowModuleValue = flow.value.modules[step].value
+
+	if (flowModuleValue.path) {
+		const moduleValue = await createInlineScriptModuleFromPath(flowModuleValue.path)
+		flowStore.update((flow: Flow) => {
+			flow.value.modules[step].value = moduleValue
+			return flow
+		})
+	}
+}
+
+export async function createScriptFromInlineScript(step: number) {
+	const flow = get(flowStore)
+	const schemas = get(schemasStore)
+	const flowModuleValue = flow.value.modules[step].value
+
+	const path = `u/flow/step-${step}-${Math.floor(Math.random() * 255)}`
+
+	const newHash = await ScriptService.createScript({
+		workspace: get(workspaceStore)!,
+		requestBody: {
+			path,
+			summary: '',
+			description: '',
+			content: flowModuleValue.content!,
+			parent_hash: undefined,
+			schema: schemas[step],
+			is_template: false,
+			language: flowModuleValue.language!
+		}
+	})
+
+	if (!newHash) {
+		return
+	}
+
+	flowStore.update((flow: Flow) => {
+		flow.value.modules[step].value = {
+			type: FlowModuleValue.type.SCRIPT,
+			path: path
+		}
+
+		return flow
+	})
+	await loadSchema(step)
+}
+
+export function removeModule(step: number) {
+	flowStore.update((flow: Flow) => {
+		flow.value.modules.splice(step, 1)
+		return flow
+	})
+}
+
+export async function copyFirstStepSchema() {
+	const flow = get(flowStore)
```

Comment:
Seems a tiny bit more elegant to me to take the flow as an argument of the function